engine-precompiles/src/secp256k1.rs

```diff
@@ -19,7 +19,9 @@ mod consts {
 // Quite a few library methods rely on this and that should be changed. This
 // should only be for precompiles.
 pub fn ecrecover(hash: H256, signature: &[u8]) -> Result<Address, ExitError> {
-    assert_eq!(signature.len(), 65);
+    if signature.len() != 65 {
```

Comment:
Magic numbers? 😄 Add please constant `SIGNATURE_LEN` near with `INPUT_LEN` into the `consts` module.